src/Core/ServerSettings.h

```diff
@@ -21,10 +21,15 @@ namespace DB
     M(UInt64, max_io_thread_pool_size, 100, "The maximum number of threads that would be used for IO operations", 0) \
     M(UInt64, max_io_thread_pool_free_size, 0, "Max free size for IO thread pool.", 0) \
     M(UInt64, io_thread_pool_queue_size, 10000, "Queue size for IO thread pool.", 0) \
+    M(UInt64, max_remote_read_network_bandwidth_for_server, 0, "The maximum speed of data exchange over the network in bytes per second for read. Zero means unlimited.", 0) \
+    M(UInt64, max_remote_write_network_bandwidth_for_server, 0, "The maximum speed of data exchange over the network in bytes per second for write. Zero means unlimited.", 0) \
+    M(UInt64, max_local_read_bandwidth_for_server, 0, "The maximum speed of local reads in bytes per second. Zero means unlimited.", 0) \
+    M(UInt64, max_local_write_bandwidth_for_server, 0, "The maximum speed of remote writes in bytes per second. Zero means unlimited.", 0) \
     M(UInt64, max_backups_io_thread_pool_size, 1000, "The maximum number of threads that would be used for IO operations for BACKUP queries", 0) \
     M(UInt64, max_backups_io_thread_pool_free_size, 0, "Max free size for backups IO thread pool.", 0) \
     M(UInt64, backups_io_thread_pool_queue_size, 0, "Queue size for backups IO thread pool.", 0) \
     M(UInt64, backup_threads, 16, "The maximum number of threads to execute BACKUP requests.", 0) \
+    M(UInt64, backup_bandwidth_for_server, 0, "The maximum read speed in bytes per second for all backups on server. Zero means unlimited.", 0) \
```

Comment:
I thought about this, and initially there was separate read/write throttling (that was simplified in f5ce5d25ac2b584f4d8d93097c82896956a334ce), but, write throttling does not make sense for BACKUPs to me, since you cannot write faster then reads, so I think that one throttler for all IO should be enough here.

Reply:
And if, by some reason, separate read/write throttling will be added, it could be added on top, i.e. with separate settings, that will have this throttler as parent, to take it into account too.

Reply:
Okay, let's stick to one throttler

Reply:
> And if, by some reason, separate read/write throttling will be added, it could be added on top, i.e. with separate settings, that will have this throttler as parent, to take it into account too.

That adding on top doesn't seem very obvious, I mean if we compare then

```
backup_read_bandwidth_for_server
backup_write_bandwidth_for_server
restore_read_bandwidth_for_server
restore_write_bandwidth_for_server
```

looks more clear than
`backup_write_bandwidth_for_server` which is a child of `backup_bandwidth_for_server` (it's difficult to understand why writing is considered as a kind of reading)

Reply:
@vitlibar I'm looking at this from different perspective.

First of all I'm not sure that it make sense to have separate throttling for read and write, because basically you want to throttle the BACKUP overall to avoid eating a lot of bandwidth from the queries that are running right now, and throttling one side will throttle the other side automatically anyway.
But maybe you can you image a situation when you need both, read and write?

Secondly, by "adding on top" I mean that it still make sense to have a total throttler for BACKUPs and creating read/write on top of it will work just like that.

>backup_write_bandwidth_for_server which is a child of backup_bandwidth_for_server (it's really difficult to understand why writing is considered as a kind of reading)

`backup_bandwidth_for_server` will not mean throttling reads, it will throttle overall backups in this case, and both read and write, will inherit throttler from it. Does that make sense?

Reply:
Probably I don't quite understand. If the same throttler is passed to both ReadBuffer and WriteBuffer it will count every copied chunk twice, won't it?

Reply:
No it won't it will simply had one common bandwidth, and that's it.

For example with `backup_bandwidth_for_server=10Mi` it will not possible for read and write to exceed this bandwidth regardless of what values for `backup_read_bandwidth_for_server`/`backup_write_bandwidth_for_server` will be set.

Reply:
Ok, let's keep the name `backup_bandwidth_for_server`